topics/data-science/tutorials/python-loops/tutorial.md

```diff
@@ -0,0 +1,528 @@
+
+Writing loops saves us time and makes sure our code is accurate, that we don't accidentally introduce a typo somewhere in the loop body.
+
+
+## Indentation
+
+The first line of the `for` loop must end with a colon, and the body must be indented.
+
+> ### {% icon tip %} Tip: Blocks in Python
+>
+> The colon at the end of the first line signals the start of a *block* of statements.
+>
+> ```
+> for x in y:
+>     print(x)
+> ```
+>
+> or
+>
+> ```
+> if x > 10:
+>     print(x)
+> ```
+>
+> or even further nesting is possible:
+>
+> ```
+> for x in y:
+>     if x > 10:
+>         print(x)
+> ```
+>
+{: .tip}
+
+The indentation is in fact, quite necessary. Notice how this fails:
+
+```python
+for number in [2, 3, 5]:
+print(number)
+```
+
+And, likewise, this:
+
+```python
+patient1 = "z2910"
+  patient2 = "y9583"
+```
+
+> ### {% icon question %} Question: Correct the errors
+>
+> ```
+> data = [1, 3, 5, 9]
+> for i in data:
+> if i < 4:
+> print(i)
+> else:
+> print(i * 2)
+> print(len(data))
+> ```
+>
+> > ### {% icon solution %} Solution
+> >
+> >
+> > ```python
+> > data = [1, 3, 5, 9]
+> > for i in data:
+> >     if i < 4:
+> >         print(i)
+> >     else:
+> >         print(i * 2)
+> > # But what about this line:
+> > print(len(data))
+> > ```
+> >
+> > Here this code is actually ambiguous, we don't know how indented `print(len(data))` should be. This very synthetic example lacks context, but there are three places it could be, with three different effects.
+> > The first option, no indentation, prints out the list length at the end.
+> >
+> > ```python
+> > [...]
+> >     else:
+> >         print(i * 2)
+> > print(len(data))
+> > ```
+> >
+> > The second, prints out the length once per loop iteration.
+> >
+> > ```python
+> > data = [1, 3, 5, 9]
+> > for i in data:
+> >     if i < 4:
+> >         print(i)
+> >     else:
+> >         print(i * 2)
+> >     print(len(data))
+> > ```
+> >
+> > First the `if` and `else` case resolve themselves, and then `len(data)` is printed every single loop iteration.
+> > The last case only prints out the length if `i >= 4`.
+> >
+> > ```python
+> > data = [1, 3, 5, 9]
+> > for i in data:
+> >     if i < 4:
+> >         print(i)
+> >     else:
+> >         print(i * 2)
+> >         print(len(data))
+> > ```
+> >
+> {: .solution}
+{: .question}
+
+```python
+# Check your answers here!
+data = [1, 3, 5, 9]
+for i in data:
+if i < 4:
+print(i)
+else:
+print(i * 2)
+print(len(data))
+```
+
+## Variable Naming
+
+Loop variables can be called anything, `i`, `j`, and `k` are very commong defaults due to their long history of use in other programing languages.
+As with all variables, loop variables are: Created on demand, and Meaningless; their names can be anything at all.
+
+```python
+for kitten in [2, 3, 5]:
+    print(kitten)
+```
+
+But *meaningless* is bad for variable names, and whenever possible, we should strive to pick useful, accurate variable names that help use remember what's going on:
```

Comment:
I love it that you addressed this. Seen unfortunately alot of weird variable names like food in student scripts